mayastor/src/bdev/nexus/nexus_io.rs

```diff
@@ -108,13 +108,33 @@ impl Bio {
 
     /// assess the IO if we need to mark it failed or ok.
     #[inline]
-    pub(crate) fn assess(&mut self) {
+    pub(crate) fn assess(
+        &mut self,
+        child_io: *const spdk_bdev_io,
+        success: bool,
+    ) {
         self.ctx_as_mut_ref().in_flight -= 1;
 
         if cfg!(debug_assertions) {
             assert_ne!(self.ctx_as_mut_ref().in_flight, -1);
         }
 
+        if !success && !child_io.is_null() {
+            let io_type = Bio::io_type(self.0).unwrap();
+            let io_offset = self.offset();
+            let io_num_blocks = self.num_blocks();
+
+            unsafe {
+                self.nexus_as_mut_ref().error_record_add(
```

Comment:
do you actually need it as mut_ref? seems like error_record_add only uses self?

Reply:
You're right. It was using mut self before my very last set of changes, but now it doesn't and I didn't spot that the mut can go. Thanks :-)